### ISSUE TITLE
swarm_test.TestDockerSwarm failed as volume is not attached

### ISSUE THREAD

**Issue by ashahi1** [vmware/docker-volume-vsphere#1490]
CI: https://ci.vmware.run/vmware/docker-volume-vsphere/739

Sam, as swarm test failed, can you please take a look at it?

I was trying to fix issue https://github.com/vmware/docker-volume-vsphere/issues/1284#issuecomment-311200955 and this test failed.

Test logs:
```
2017/06/27 03:11:08 START: swarm_test.TestDockerSwarm
2017/06/27 03:11:08 Fetching full name for volume [swarm_test_volume_184158] from VM [192.168.31.81]
2017/06/27 03:11:09 Full volume name: [swarm_test_volume_184158@datastore1]
2017/06/27 03:11:09 Creating a docker service [swarm_test_service_239730] on VM [192.168.31.81]
2017/06/27 03:11:10 Checking docker service [swarm_test_service_239730] status on VM [192.168.31.81]
2017/06/27 03:11:10 Listing docker service [swarm_test_service_239730] running on VM [192.168.31.81]
2017/06/27 03:11:10 Checking running containers for docker service [swarm_test_service_239730] on docker hosts: [192.168.31.81 192.168.31.85 192.168.31.83]
2017/06/27 03:11:11 Sleep for 5 seconds
2017/06/27 03:11:23 Containers running on docker host [192.168.31.83]: swarm_test_service_239730.1.lww2tro7aw22uh57s0pq4bbq6
2017/06/27 03:11:23 Confirming attached status for volume [swarm_test_volume_184158]
2017/06/27 03:11:27 Expected Attached VM name is [photon.vmfs]
2017/06/27 03:11:27 Attached VM name from Docker CLI is [Ubuntu.14.04]
2017/06/27 03:11:27 Attached VM name from Admin CLI is [Ubuntu.14.04]

----------------------------------------------------------------------
FAIL: swarm_test.go:91: SwarmTestSuite.TestDockerSwarm

swarm_test.go:106:
    c.Assert(status, Equals, true, Commentf("Volume %s is not attached", s.volumeName))
... obtained bool = false
... expected bool = true
... Volume swarm_test_volume_184158 is not attached

2017/06/27 03:11:27 Destroying volume [swarm_test_volume_184158]

----------------------------------------------------------------------
FAIL: swarm_test.go:65: SwarmTestSuite.TearDownSuite

swarm_test.go:68:
    c.Assert(err, IsNil, Commentf(out))
... value *exec.ExitError = &exec.ExitError{ProcessState:(*os.ProcessState)(0xc4201f3ea0), Stderr:[]uint8(nil)} ("exit status 1")
... Error response from daemon: unable to remove volume: remove swarm_test_volume_184158: VolumeDriver.Remove: Failed to remove volume swarm_test_volume_184158, in use by VM = Ubuntu.14.04.
```

**Comment by shuklanirdesh82** [comment]
```
2017/06/27 03:11:27 Expected Attached VM name is [photon.vmfs]
2017/06/27 03:11:27 Attached VM name from Docker CLI is [Ubuntu.14.04]
2017/06/27 03:11:27 Attached VM name from Admin CLI is [Ubuntu.14.04]
```

I guess test needs to fix the expectation here for the swarm test. Test is expecting `s.master` always and needs to be passed correct expectation.

**Comment by ashahi1** [comment]
Making it P0 as it is failing on CI

**Comment by shaominchen** [comment]
@ashahi1 Are you sure this is not related to the change itself? I guess it's still the same issue as before and that's why those code were commented out. The CI after yours have already passed: https://ci.vmware.run/vmware/docker-volume-vsphere/746

I'm closing this as this issue is really unrelated to swarm test. The test depends on util function VerifyAttachedStatus, which again depends on the RetrieveVMNameFromIP. Please investigate further (about the issue with RetrieveVMNameFromIP) before uncommenting it.

**Comment by ashahi1** [comment]
@shaominchen To me it looks to be a test issue. Please see the output below. Your test is expecting a different vm name. It has nothing to do with RetrieveVMNameFromIP. 
We are getting vm name it just that comparison fails because test expects name as photon.vmfs but colume is attached to different vm (Ubuntu.14.04).

We had commented out because govc was not returning the name. After uncommenting, swarm test fails, so swarm test needs to be fixed here.

Re-opening this issue.

```
2017/06/27 03:11:27 Expected Attached VM name is [photon.vmfs]   <<<<<<<<<<<<<<<<<
2017/06/27 03:11:27 Attached VM name from Docker CLI is [Ubuntu.14.04]
2017/06/27 03:11:27 Attached VM name from Admin CLI is [Ubuntu.14.04]
```

**Comment by shaominchen** [comment]
@ashahi1 Yes, this is a test issue. It is exposed by re-enabling the RetrieveVMNameFromIP logic in VerifyAttachedStatus. The fix is just trivial:

102: status, _ = verification.IsDockerContainerRunning(s.swarmNodes, s.serviceName, 1)
==> status, host := verification.IsDockerContainerRunning(s.swarmNodes, s.serviceName, 1)

105: status = verification.VerifyAttachedStatus(s.volumeName, s.master, s.esxName)
==> status = verification.VerifyAttachedStatus(s.volumeName, host, s.esxName)

114: status, host := verification.IsDockerContainerRunning(s.swarmNodes, s.serviceName, 2)
==> status, host = verification.IsDockerContainerRunning(s.swarmNodes, s.serviceName, 2)

You can include above fix in your current PR (instead of disabling the swarm test), or otherwise I will send out a separate fix after your PR is in.

**Comment by ashahi1** [comment]
@shaominchen I tried the steps that you suggested but still the test fails. Test is expecting volume to be always attached to the master node.  There are multiple instances where we do verification based on assumption that operation executed on master node only - this should be fixed.

**Comment by shaominchen** [comment]
Fixed with PR #1492.
